lib/macho/macho_file.rb

```diff
@@ -189,7 +189,38 @@ def add_command(lc)
       set_sizeofcmds(new_sizeofcmds)
       @raw_data[header.class.bytesize + sizeofcmds, cmd_raw.bytesize] = cmd_raw
 
-      # synchronize fields with the raw data
+      synchronize_fields if synchronize
+    end
+
+    # Delete a load command from the Mach-O.
+    # @param lc [MachO::LoadCommand] the load command being deleted
+    # @param synchronize [Boolean] whether or not to synchronize the instance fields
+    # @return [void]
+    # @note This is public, but methods like {#delete_rpath} should be preferred.
+    #  Setting `synchronize` to false **will break the instance** unless
+    #  {#synchronize_fields} is called **immediately** afterwards.
+    def delete_command(lc, synchronize = true)
+      @raw_data.slice!(lc.view.offset, lc.cmdsize)
+
+      # decrease the number of load commands by the number deleted
+      set_ncmds(ncmds - 1)
+
+      # decrease the total size of commands by the size of the deleted commands
+      set_sizeofcmds(sizeofcmds - lc.cmdsize)
```

Comment:
I think you can be a bit less verbose with this and the previous non-empty/non-comment line:

```ruby
# Update Mach-O header fields to account for deleted load command.
set_ncmds(ncmds - 1)
set_sizeofcmds(sizeofcmds - lc.cmdsize)
```

(The rest is already evident from the code itself and the surrounding method.)